modules/apps/27-interchain-accounts/host/keeper/migrations.go

```diff
@@ -22,10 +22,13 @@ func NewMigrator(k *Keeper) Migrator {
 func (m Migrator) MigrateParams(ctx sdk.Context) error {
 	if m.keeper != nil {
 		var params types.Params
-		m.keeper.legacySubspace.GetParamSet(ctx, &params)
-
-		if err := params.Validate(); err != nil {
-			return err
+		if m.keeper.legacySubspace == nil {
```

Comment:
Just to clarify when icahost was added in your upgrade it was created with a `nil` legacy subspace? @mmsqe

Reply:
Okay, thank you! In that case this should be fine.

I'm curious about a case in which an actual params subspace is created and passed but which does not have any params stored in the param set from a previous version. As genesis would already have been run for the ica module but not setup default params I would imagine it to be empty. I assume in that case `params.Validate()` should error below, and this would technically misconfiguration by the chain's app wiring.

Reply:
Aha, yea I believe you're right @DimitrisJim. With `RegisterParamSet(&Params{})`, right? 

That's probably fine behaviour imo - given this would probably be the result of misconfig anyways - a param's change would need to be done for allowing msgs in `[]string` then. This is a pretty edge case scenario either way. Most chains deploy host first anyways, and controller later

Reply:
Wondering if this migration code would be safer and cover edge cases such as described above (this would at least init params with all msgs allowed instead of the empty string slice, its a bit more explicit than relying on a side effect of `RegisterParamSet`):

```go
if m.keeper != nil {
        params := types.DefaultParams()
	if m.keeper.legacySubspace != nil {
		m.keeper.legacySubspace.GetParamSetIfExists(ctx, &params)
	}

	if err := params.Validate(); err != nil {
		return err
	}

	m.keeper.SetParams(ctx, params)
	m.keeper.Logger(ctx).Info("successfully migrated ica/host submodule to self-manage params")
}
```

Any thoughts? Happy to stick with the current changes tho if agreed.

Reply:
> I'd assume we'd want to panic if we try to get something and it doesn't exist.

I'm not sure, and I don't feel particularly strongly about this - if it doesn't exist we've at least assigned safe defaults via `types.DefaultParams` rather than panicking. Do you have any feedback here @mmsqe, as someone maintaining a chain?

> One sidenote here, should we be doing same sanity checks for controller?

@mmsqe yes I think it would be nice to align the controller migration code as well, thank you! 🙏